tests/TaskTests/SecretsManagerGetSecret/SecretsMangerGetSecret-test.ts

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { SecretsManager } from 'aws-sdk'
+import { SdkUtils } from 'Common/sdkutils'
+import { TaskOperations } from '../../../Tasks/SecretsManagerGetSecret/GetSecretTaskOperations'
+import { TaskParameters } from '../../../Tasks/SecretsManagerGetSecret/GetSecretTaskParameters'
+
+// unsafe any's is how jest mocking works, so this needs to be disabled for all test files
+// tslint:disable: no-unsafe-any
+jest.mock('aws-sdk')
+
+const defaultTaskParameters: TaskParameters = {
+    awsConnectionParameters: undefined,
+    secretIdOrName: undefined,
+    variableName: 'secret',
+    versionId: undefined,
+    versionStage: undefined
+}
+
+const secretsManagerFails = {
+    promise: function() { throw new Error('doesn\'t exist') }
+}
+
+const secretsManagerReturnsString = {
+    promise: function() { return {SecretString: 'string'} }
+}
+
+const secretsManagerReturnsBadBase64 = {
+    promise: function() { return {SecretBinary: 'strOo0ing'} }
+}
+
+const secretsManagerReturnsValidBase64 = {
+    promise: function() { return {SecretBinary: 'YWJjCg=='} }
+}
+
+const assertAbc = {
+    promise: function(request: any) {
+        expect(request.VersionId).toBe('abc')
+        expect(request.versionStage).toBe(undefined)
+    }
+}
+
+describe('Secrets Manger Get Secret', () => {
+    // TODO https://github.com/aws/aws-vsts-tools/issues/167
+    beforeAll(() => {
+        SdkUtils.readResourcesFromRelativePath('../../../_build/Tasks/SecretsManagerGetSecret/task.json')
+    })
+
+    test('Creates a TaskOperation', () => {
+        expect(new TaskOperations(new SecretsManager(), defaultTaskParameters)).not.toBeNull()
+    })
+
+    test('Fails on exception thrown', async () => {
+        const badSecretsManager = new SecretsManager() as any
+        badSecretsManager.getSecretValue = jest.fn((params: any, cb: any) => secretsManagerFails)
+        const taskOperations = new TaskOperations(badSecretsManager, defaultTaskParameters)
+        await taskOperations.execute().catch((e) => { expect(e.message).toContain('doesn\'t exist') })
+    })
+
+    test('Fails on bad base64', async () => {
+        const badSecretsManager = new SecretsManager() as any
+        badSecretsManager.getSecretValue = jest.fn((params: any, cb: any) => secretsManagerReturnsBadBase64)
+        const taskOperations = new TaskOperations(badSecretsManager, defaultTaskParameters)
+        await taskOperations.execute().catch((e) => {
+            expect(e.message).toContain('the string to be decoded is not correctly encoded')
+        })
+    })
+
+    test('Handles secret string', async () => {
+        const badSecretsManager = new SecretsManager() as any
```

Comment:
* Missing `assertAbc` in a few calls
* Happy path methods call client `badSecretsManager`

Reply:
Added a comment about why we can't do that with the current project structure (sadly) and fixed the client name in the happy path